### DISCUSSION TITLE
DXMT September Status: To a New Level of Completeness

### DISCUSSION THREAD

**Discussion by 3Shain** [3Shain/dxmt#19] [Announcements]
`DXMT` is a Metal-based translation layer for Direct3D 11. It currently allows running D3D11 applications on macOS with the aid of `Wine`.

We are happy to show the new stage we've reached in DXMT development since the last announcement.

## New Progresses & Improvements

<img align="left" width="70%" src="https://github.com/user-attachments/assets/59e593de-94c4-4c11-a3ce-3b10b9990eb5">

Tessellation is now largely supported. It's a new feature brought by DirectX 11 which converts low-detail subdivision surfaces into higher-detail primitives on the GPU.
<br><br><br>

<img  align="right"  width="70%" src="https://github.com/user-attachments/assets/d3852498-5474-47f2-9df6-b487a229fc33">
<br>

> Tessellation is usually applied in games for footprints, water waves, etc.

The challenge is, DirectX tessellation model is only partially supported in Metal. Thankfully we have managed to tackle it by mesh pipeline emulation and the implementation is fairly efficient and performant. It was a large amount of work in both shader converter and API implementation side.

---

Besides tessellation we have also brought: 
- Better blit command implementation
- Initial multisample support
- Debugging infrastructure
- UAV counter
- Dual-source blending
- Shader converter (airconv) improvements
- Targeting DirectX feature level 11.1 (with known restrictions)
- Many corner case and bug fixes, as well as performance improvements

As for now we have got an even more apparent picture of our next steps. A complete roadmap (a.k.a. TODOs before release) is planned to be revealed in mid/late-October.

## For Geeks: Try it and see

If you're familiar with Wine setup (not by a wrapper program like CrossOver/Whisky, it is about terminal commands stuff), we have got [instructions to test out DXMT](https://github.com/3Shain/dxmt/wiki/DXMT-Installation-Guide-for-Geeks) for you and expect any feedback for you (and you can help us collect information of supported/unsupported games). _Please restrict the reports as well as enquiries in the Discord server or in this repository. Any other platform/community is not monitored._ And __it is not a product release, don't expect too much!__

<img width="49%"  src="https://github.com/user-attachments/assets/d429e253-8cc8-407e-bd11-365756386c10">

<img width="49%"  src="https://github.com/user-attachments/assets/9a4617be-70c2-4906-ab98-72ee9e31d959">

> Recently we are testing some Source 2 Engine titles! Also in the near future, the CodeWeavers QA team will help us evaluate the performance of DXMT in many games, _probably more than you can think of._

**Comment by ScathachSkadi** [comment]
I love you 3Shain 😘

**Reply by Lifeisawful** [reply]
Very impressive work! Thank you for sharing an update!

**Comment by oscarbg** [comment]
Nice..
Comparison vs:
1)native wine dx11 vulkan backend+ moltenvk 
2)dxvk+moltenvk
3)gptk 2.0 dx11 driver 
in regards to perf and completeness api wise of dx11 feature set

**Comment by ZeppLu** [comment]
So how do I join discord? The link I found in wiki seems expired

**Reply by 3Shain** [reply]
Updated.
